portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -58,21 +76,46 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get name and text
-    String name = request.getParameter("name-input");
-    String text = request.getParameter("text-input");
-    if (text != null && name != null) {
-      long timestamp = System.currentTimeMillis();
-
-      Entity messageEntity = new Entity(Constants.MESSAGE_ENTITY_TYPE);
-      messageEntity.setProperty("name", name);
-      messageEntity.setProperty("text", text);
-      messageEntity.setProperty("timestamp", timestamp);
-
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      datastore.put(messageEntity);
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+        String userEmail = userService.getCurrentUser().getEmail();
+        
+        // Get name and text
+        String name = getUserNickname(userService.getCurrentUser().getUserId());
+        String text = request.getParameter("text-input");
+        if (text != null && name != null) {
+        long timestamp = System.currentTimeMillis();
+
+        Entity messageEntity = new Entity(Constants.MESSAGE_ENTITY_TYPE);
+        messageEntity.setProperty("name", name);
+        messageEntity.setProperty("text", text);
+        messageEntity.setProperty("email", userEmail);
+        messageEntity.setProperty("timestamp", timestamp);
```

Comment:
Since you're so far on Week 4, let's add some experience with something that will make your life WAY easier once we start the project.

See if you can use objectify to do persistence, rather than using this property based storage. It'll be much harder at first, and then after you have it set up, it'll be trivial to add or remove stored features.

Reply:
You can find their documentation here: https://github.com/objectify/objectify